mavros_extras/src/plugins/fake_gps.cpp

```diff
@@ -0,0 +1,309 @@
+			earth: GeographicLib::Constants::WGS84_a(),
+			GeographicLib::Constants::WGS84_f();
+			/**
+			 * @brief Conversion of the origin from geodetic coordinates (LLA)
+			 * to ECEF (Earth-Centered, Earth-Fixed)
+			 */
+			earth.Forward(map_origin.x(), map_origin.x(), map_origin.x(),
+						ecef_origin.x(), ecef_origin.y(), ecef_origin.z());
+		}
+		catch (const std::exception& e) {
+			ROS_INFO_STREAM("FGPS: Caught exception: " << e.what() << std::endl);
+		}
+
+		// source set params
+		fp_nh.param("use_mocap", use_mocap, true);		// listen to MoCap source
+		fp_nh.param("mocap_transform", mocap_transform, true);	// listen to MoCap source (TransformStamped if true; PoseStamped if false)
+		fp_nh.param("tf/listen", tf_listen, false);		// listen to TF source
+		fp_nh.param("use_vision", use_vision, false);		// listen to Vision source
+
+		// tf params
+		fp_nh.param<std::string>("tf/frame_id", tf_frame_id, "map");
+		fp_nh.param<std::string>("tf/child_frame_id", tf_child_frame_id, "fix");
+		fp_nh.param("tf/rate_limit", tf_rate, 50.0);
+
+		if (use_mocap) {
+			if (mocap_transform) {	// MoCap data in TransformStamped msg
+				mocap_tf_sub = fp_nh.subscribe("mocap/tf", 10, &FakeGPSPlugin::mocap_tf_cb, this);
+			}
+			else {	// MoCap data in PoseStamped msg
+				mocap_pose_sub = fp_nh.subscribe("mocap/pose", 10, &FakeGPSPlugin::mocap_pose_cb, this);
+			}
+		}
+		else if (use_vision) {	// Vision data in PoseStamped msg
+			vision_pose_sub = fp_nh.subscribe("vision", 10, &FakeGPSPlugin::vision_cb, this);
+		}
+		else if (tf_listen) {	// Pose aquired from TF Listener
+			ROS_INFO_STREAM_NAMED("fake_gps", "Listen to transform " << tf_frame_id
+										 << " -> " << tf_child_frame_id);
+			tf2_start("FakeGPSVisionTF", &FakeGPSPlugin::transform_cb);
+		}
+		else {
+			ROS_ERROR_NAMED("fake_gps", "No pose source!");
+		}
+	}
+
+	Subscriptions get_subscriptions()
+	{
+		return { /* Rx disabled */ };
+	}
+
+private:
+	friend class TF2ListenerMixin;
+	ros::NodeHandle fp_nh;
+
+	// Constructor for a ellipsoid
+	GeographicLib::Geocentric earth;
+
+	ros::Subscriber mocap_tf_sub;
+	ros::Subscriber mocap_pose_sub;
+	ros::Subscriber vision_pose_sub;
+
+	bool use_mocap;			//!< set use of mocap data (PoseStamped msg)
+	bool use_vision;		//!< set use of vision data
+	bool mocap_transform;		//!< set use of mocap data (TransformStamped msg)
+	bool tf_listen;			//!< set use of TF Listener data
+
+	double eph, epv;
+	int satellites_visible;
+	GPS_FIX_TYPE fix_type;
+
+	double tf_rate;
+	std::string tf_frame_id;
+	std::string tf_child_frame_id;
+	ros::Time last_transform_stamp;
+
+	ros::Time last_pos_time;
+	ros::Duration gps_period;
+
+	Eigen::Vector3d map_origin;	//!< geodetic origin [lla]
+	Eigen::Vector3d ecef_origin;	//!< geocentric origin [m]
+	Eigen::Vector3d old_ecef;	//!< previous geocentric position [m]
+	double old_stamp;		//!< previous stamp [s]
+
+	inline Eigen::Vector3d enu_to_ecef_transform(const Eigen::Affine3d &map_point)
```

Comment:
Is the same as the inverse of the other rotation matrix basically. So I suppose it can be in a header file. Though, since this is a coordinate frame transform (both are local coordinates), this can go to the same place as the other frame transforms are. What do you think? We keep it here or move it?